### ISSUE TITLE
Combat view

### ISSUE THREAD

**Issue by AideeAmei** [CleverRaven/Cataclysm-DDA#78283]
### Is your feature request related to a problem? Please describe.

Not sure if this is already another FR, someone else might call it different names and such.

Imagine Danny Rivers, a thief and a scoundrel, likes to watch them zombies from afar while going about his raiding business on the brink of sundown. Moving the view west helps this a lot in the Lycra st. on the east side of Hillburry.

However, the Z's might move a bit north or so while he's lockpicking the s*** out of the metal door and the view might not capture their movement. Danny, focused on getting into the pawnshop, might lose track of some tangos while doing stuff.

### Solution you would like.

ENTER THE MIGHTY POWER OF COMBAT VIEW

Select one or more enemies, and the view will shift accordingly, capturing all of them:
Danny and zombie child and the feral predator eating a panicked person in the food truck.
The view is focused on a tile in their midst. The calculation for the tile coordinates can include weight for the player character, that shifts the view slightly more towards them.

"Okay, Aidee, but is that really all?" you say.
No! The combat view also has an option to zoom out and in automatically if needed to cover all targets!
Two fascinating functionalities for a price of one!

### Describe alternatives you have considered.

Danny already is a master of manual view moving and has excellent coordination with his powerful key bindings having view moves and zoom near each other, but that doesn't benefit Lee, a 19 years old player from Ontario, that only has one arm and moving view gets complicated for him. Lee even tried to play with mouse to see if that helped but it sucked for him because he's right handed and only has left arm.

### Additional context

I just had to write it this way. Sorry.

**Comment by CoroNaut** [comment]
I like the idea, but I dont think its for CDDA. If this combat view had an option to zoom and adjust to contain ALL enemies and allies in the view, every second that passes will adjust the entire view. I can see this can become very disorienting very quickly. Even if we consider the case where you track just yourself and one other zombie, the view and zoom will adjust as you two approach each other every second. Once again, this can get disorienting quickly. Games that have this sort of feature often come with smooth camera movements for a reason. CDDA is an ascii/tiles game, so every movement of the camera will be in very large steps and be snappy. 

How about if that zombie you track goes behind a building? The view will instantly snap to just the player, and then you would probably have to re-track that zombie if it comes back into view (after you zoom out from this combat view mode). If it came back into view and it's still tracked, your entire view will zoom out and adjust all of the sudden, leaving you to wonder what the heck happened. 

Not to mention, the view adjusting to contain all your enemies will likely have the player near the edge of the view. If the character is only 10 tiles from the bottom of your screen, and you want to go south, the player can only see those 10 tiles. Sure you would know where your enemies are, but what if a new un-tracked zombie comes at you from the south? It will only be seen when its within 10 tiles (if you are even looking at the bottom of your screen).

Then theres the issue of long tasks that pass time 5/30/60 minutes at a time. combat view can't work unless a frame is drawn every second. if they were, and combat view was on, the players camera would be zipping and zooming as full in-game minutes pass every real-life second. Lockpicking in your example is a longer task, so not every frame for every second is drawn. So combat view would have to be disabled, or combat view would have to slow down the game to real-time to keep combat view working as intended.

**Comment by ZhilkinSerg** [comment]
This won't add anything useful for a player.

**Comment by IdleSol** [comment]
I still don't understand what you're suggesting. (Difficulties with translation).

Move the center of the bubble? Use `shift + h` or `J`, `K`, `L` and `:` - to return the default view (centered on the character's position). That way you can move the bubble in the right direction. Although I don't see why that would be necessary.

@CoroNaut

Theoretically. A sniper rifle (binoculars and analogs) could have remote viewing functionality. Say within a radius of 2 to 5 omt (altitude dependent? ~ +1 for each level above 2?). And would take some time ~10 seconds (debatable).

And it would work by shifting the center of the field of view from the character to the selected omt. Then a new bubble of <s>reality</s> visibility would be drawn from these coordinates.

That is, from the player's point of view:
1. Pick up an item with the `zoom` flag
2. activate the item
3. jump to the world map
4. select omt for review or cancel it
5. receive a message: character ... is using ... and is focusing on an object in the distance
6. perform a 10 second delay
7. change the view from the reality bubble to a new bubble centered on the selected omt
8. functionality available: skip move, cancel view, inspect things, zombies and tiles (`.`, `x`, `V`, `z`/`Z`)

The game sequentially processes two bubbles: the first is the reality bubble, the second is the view bubble. Ideally, a transparency cache should be applied to the view bubble, taking into account that the character is not in the center, but outside the bubble and on a certain side (is this even possible?).

Theoretically #2. It would be possible to additionally implement the ability to shoot from sniper weapons. And thus kill serious enemies, without the need to approach in close combat. Still, 60 meters is a distance for a pistol, not a sniper rifle. (Penalties in sound and change the direction of movement of all hordes in a certain radius).

**Comment by AideeAmei** [comment]
> I still don't understand what you're suggesting. (Difficulties with translation).
> 
> Move the center of the bubble?

No, move the view (view is not the bubble, see key bindings screen for basic in-game), like you suggested, HJKL etc. or whatever your bindings are, is the **manual way**, what I suggest is targeting and marking a few enemies and move the view each turn automatically depending on their position and your position. Manual move view is super and all, but when the concentration is focused on moving the view, it's shifted off actual issues.

**Comment by AideeAmei** [comment]
> This won't add anything useful for a player.

Seems to me "move view" didn't either, then. Or Advanced Inventory, for that part. You can pick up and drop, no? You can drop directionally, pick up from all 9 tiles, why another screen?

Or why should we have auto pick and other stuff? And why aiming can snap to target? Isn't that a similar thing? And totally useless, too?

I get it, I don't like auto forage, I don't use it, but hey, it's a great functionality and I totally love it's an option should I ever reconsider. I think it's the same with auto move view.

**Comment by ZhilkinSerg** [comment]
> > This won't add anything useful for a player.
> 
> Seems to me "move view" didn't either, then. Or Advanced Inventory, for that part. You can pick up and drop, no? You can drop directionally, pick up from all 9 tiles, why another screen?
> 
> Or why should we have auto pick and other stuff? And why aiming can snap to target? Isn't that a similar thing? And totally useless, too?
> 
> I get it, I don't like auto forage, I don't use it, but hey, it's a great functionality and I totally love it's an option should I ever reconsider. I think it's the same with auto move view.

Your examples are completely irrelevant

**Comment by AideeAmei** [comment]
> > > This won't add anything useful for a player.
> > 
> > 
> > Seems to me "move view" didn't either, then. Or Advanced Inventory, for that part. You can pick up and drop, no? You can drop directionally, pick up from all 9 tiles, why another screen?
> > Or why should we have auto pick and other stuff? And why aiming can snap to target? Isn't that a similar thing? And totally useless, too?
> > I get it, I don't like auto forage, I don't use it, but hey, it's a great functionality and I totally love it's an option should I ever reconsider. I think it's the same with auto move view.
> 
> Your examples are completely irrelevant

You opinion is valued. Thank you for it. I still think you're wrong, but I do so with love and compassion.
Nothing much more to say, I don't think they're wrong, I think they're completely relevant and basically describing the same situations in different areas.

If you don't understand, you can ask. If you have an opinion you care to explain, you can do so. Nobody is stopping you, nobody is censoring you. But you have to say more than "irrelevant" or "not useful". These comments of yours have not been helpful. They're cryptic.
